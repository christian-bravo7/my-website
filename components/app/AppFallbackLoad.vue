<template>
  <div
    class="fixed inset-0 z-50 flex justify-center items-center bg-white text-pink-800 dark:bg-blue-600 dark:text-blue-500 text-12xl"
  >
    <SiteLogo class="fill-current pulse-effect" />
  </div>
</template>

<script lang="ts">
import { Vue, Component } from 'nuxt-property-decorator';

import SiteLogo from '~/assets/img/logo.svg?inline';

@Component({
  components: {
    SiteLogo,
  },
})
export default class AppFallbackLoad extends Vue {}
</script>

<style lang="scss" scoped>
.pulse-effect {
  animation-name: pulse;
  animation-duration: 500ms;
  animation-timing-function: ease-in-out;
  animation-iteration-count: infinite;
  animation-direction: alternate;
}

@keyframes pulse {
  0% {
    transform: scale(1);
    opacity: 1;
  }

  100% {
    transform: scale(0.9);
    opacity: 0.5;
  }
}
</style>
