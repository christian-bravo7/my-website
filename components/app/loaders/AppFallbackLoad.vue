<template>
  <div
    class="fixed | inset-0 z-50 | flex justify-center items-center | primary-text bg-white dark:bg-dark-theme-800 "
  >
    <div class="w-64 h-64 sm:w-40 sm:h-40">
      <SiteLogo class="w-full h-full | pulse-effect" />
    </div>
  </div>
</template>

<script lang="ts">
import { Vue, Component } from 'nuxt-property-decorator';

import SiteLogo from '~/assets/img/logo.svg?inline';

@Component({
  components: {
    SiteLogo,
  },
})
export default class AppFallbackLoad extends Vue {}
</script>

<style lang="scss" scoped>
.pulse-effect {
  animation-name: pulse;
  animation-duration: 500ms;
  animation-timing-function: ease-in-out;
  animation-iteration-count: infinite;
  animation-direction: alternate;
}

@keyframes pulse {
  0% {
    transform: scale(1);
    opacity: 1;
  }

  100% {
    transform: scale(0.9);
    opacity: 0.5;
  }
}
</style>
