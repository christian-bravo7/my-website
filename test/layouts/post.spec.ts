import Vuex from 'vuex';
import PortalVue from 'portal-vue';
import { createLocalVue, mount } from '@vue/test-utils';

import Post from '@/layouts/post.vue';
import AppFallbackLoad from '@/components/app/AppFallbackLoad.vue';

const localVue = createLocalVue();
localVue.use(PortalVue);
localVue.use(Vuex);

describe('post layout', () => {
  let getters: any;
  let store: any;

  beforeEach(() => {
    getters = {
      'guest/isReady': jest.fn(() => false),
    };
    store = new Vuex.Store({
      getters,
    });
  });

  it('should render correclty', () => {
    const wrapper = mount(Post, {
      store,
      localVue,
      components: {
        AppFallbackLoad,
      },
      stubs: {
        ClientOnly: {
          template: '<div></div>',
        },
        Nuxt: {
          template: '<div></div>',
        },
      },
    });

    expect(wrapper.html()).toMatchSnapshot();
  });
});
