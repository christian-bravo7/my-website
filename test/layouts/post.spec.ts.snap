// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`post layout should render correclty 1`] = `
<main class="bg-gray-50 dark:bg-blue-600">
  <div class="fixed inset-0 z-50 flex justify-center items-center bg-white text-pink-800 dark:bg-blue-600 dark:text-blue-500 text-12xl"><svg class="fill-current pulse-effect"></svg></div>
  <div></div>
  <header>
    <!---->
  </header>
  <div></div>
</main>
`;
